Synview-frontend/src/components/SyncForm.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useState } from "react";
+import { useGetMyCommitsMutation } from "../services/apiSlice.ts";
+import { useAppSelector } from "../hooks.ts";
+
+export default function SyncForm() {
+  const [githubUsername, setGithubUsername] = useState("");
+  const [projectName, setProjectName] = useState("");
+  const [getCommits] = useGetMyCommitsMutation();
+  const project_id = useAppSelector((state) => state.githubModal.project_id);
+  const user_id = useAppSelector((state) => state.githubModal.user_id);
+
+  const handleSyncForm = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    try {
+      if (projectName && githubUsername && project_id && user_id) {
```

Comment:
You don't handle the else case here. Often you'd want a wrapper around elements like `input` or `select` to support a full state model, including whether it's errored. Then you would include a validator either for the form or on the fields. That way if the user tries to submit, the invalid fields are highlighted with relevant errors. [react-hook-form](https://react-hook-form.com/get-started) is library that provides this functionality. You can use their native syntax for validation or even use [a zod plugin](https://www.npmjs.com/package/@hookform/resolvers).